packages/core/saved-objects/core-saved-objects-api-server/src/apis/bulk_create.ts

```diff
@@ -6,10 +6,8 @@
  * Side Public License, v 1.
  */
 
-import type {
-  SavedObjectReference,
-  SavedObjectsMigrationVersion,
-} from '@kbn/core-saved-objects-common';
+import type { SavedObjectsMigrationVersion } from '@kbn/core-saved-objects-common';
+import type { SavedObjectReference } from '@kbn/core-saved-objects-common/src/server_types';
```

Comment:
Should we be using deep imports? It's not a pattern we typically follow in the core packages. We prefer root-level imports where possible. Is there a way to expose these at the root level rather? how about using type aliases instead?

It's a tricky one though since we're handling both legacy common types and server types.

Reply:
Yeah, I hear what you're saying. I've realised that there is a slightly tricky situation that arose from https://github.com/elastic/kibana/pull/148979 and it has to do with circular kbn project references.

I assumed that we are moving `SavedObject` (and related types) from `core-saved-objects-common` to `core-saved-objects-server`. What I have since realised is that `SavedObject` and related types are also used by `core-saved-objects-api-server`. By having the new home be in `core-saved-objects-server` a circular dependency now exists between `api-server` and `server` packages.

What I'm now realising is that the new home of `SavedObject` should probably be `api-server` package instead since the `server` package depends on it. I'll restructure that code here to reflect this change. In this way we can "trade" multiple anti-pattern imports for just one at the top-level of the `core-saved-objects-api-server` package. We should be able to remove that import once we actually move `server_types` out of `common.

Let me know if you want to chat about any of this as I may not have done the best job explaining the issue.